src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_FindReferences_Legacy.cs

```diff
@@ -62,19 +59,35 @@ public static partial class SymbolFinder
         /// information as the search is undertaken.</param>
         /// <param name="documents">An optional set of documents to be searched. If documents is null, then that means "all documents".</param>
         /// <param name="cancellationToken">An optional cancellation token.</param>
-        public static async Task<IEnumerable<ReferencedSymbol>> FindReferencesAsync(
+        public static Task<IEnumerable<ReferencedSymbol>> FindReferencesAsync(
             ISymbol symbol,
             Solution solution,
             IFindReferencesProgress progress,
             IImmutableSet<Document> documents,
             CancellationToken cancellationToken = default)
+        {
+            return FindReferencesAsync(
+                symbol, solution, progress, documents,
+                FindReferencesSearchOptions.Default, cancellationToken);
+        }
+
+        // Internal so both our existing API can call this *as well as* test code that wants
+        // to validate options (but doesn't want to bother with all the progress-goop).
```

Comment:
❕ Use in test code is not a reason to make the method internal.

* If `internal` is needed for callers in production code, the reference to use in test code is irrelevant and should be removed
* If `internal` is *not* needed for callers in production code, decrease the accessibility to meet the needs of production code and apply the `TestAccessor` technique from #28967 to avoid breaking encapsulation

Reply:
I'm ok following this pattern... except that i don't understand it :)   I'm not sure how i would apply it here.  If more info could be provided, that would be helpful.

Reply:
Suppose the production code would compile with this method `private`, but tests are calling it so it's increased to `internal`. A better approach to the same situation would be leaving this method `private` and adding the following:

```csharp
internal static class TestAccessor
{
  internal static async Task<IEnumerable<ReferencedSymbol>> FindReferencesAsync(...)
    => SymbolFinder.FindReferencesAsync(...);
}
```

Reply:
Done.  thanks!